src/game.rs

```diff
@@ -79,19 +81,120 @@ impl Game {
+        // I want anyways
+        if (wall == WallType::PurpleDoor && color == HeisterColor::Purple)
+            || (wall == WallType::OrangeDoor && color == HeisterColor::Orange)
+            || (wall == WallType::YellowDoor && color == HeisterColor::Yellow)
+            || (wall == WallType::GreenDoor && color == HeisterColor::Green)
+        {
+            return MoveValidity::Valid;
+        } else {
+            MoveValidity::Invalid("Can't move heister through wrong-colored door".to_string())
+        }
+    }
+
+    fn process_move(&mut self, m: Move) -> MoveValidity {
         let grid = self.get_absolute_grid();
-        let my_square = match grid.get(&pos) {
-            Some(my_square) => my_square,
-            None => return MoveValidity::Invalid("Square {:?} doesn't exist".to_string()),
+
+        let heister_color = m.heister_color;
+        let heister = self.get_heister_from_vec(heister_color.clone()).unwrap();
+        let heister_pos = &heister.map_position;
+
+        let dest_pos = m.position;
+        match grid.get(&dest_pos) {
+            None => {
+                return MoveValidity::Invalid(format!(
+                    "Destination square {:?} doesn't exist",
+                    dest_pos
+                ))
+            }
+            Some(_wildcard) => (),
         };
-        MoveValidity::Valid
+        // OK - if the squares are adjacent, then we can assume they're trying to
+        // move to an adjacent square, and can check for doors/walls
+        if Self::are_adjacent(heister_pos.clone(), dest_pos.clone()) {
+            // Is the move valid for the wall between these two squares?
+            // NOTE: I'm only going to check the wall of the source square -
+            // edge cases where dest square wall may not match, but for now, don't are
+            let heister_square = match grid.get(&heister_pos) {
+                Some(s) => s,
+                None => {
+                    return MoveValidity::Invalid(format!(
+                        "Heister square {:?} doesn't exist",
+                        heister_pos
+                    ))
+                }
+            };
+            let move_dir = Self::adjacent_move_direction(heister_pos.clone(), dest_pos.clone());
+            let blocking_wall = match move_dir {
+                MoveDirection::North => heister_square.north_wall,
+                MoveDirection::East => heister_square.east_wall,
+                MoveDirection::South => heister_square.south_wall,
+                MoveDirection::West => heister_square.west_wall,
+            };
+            let validity = match blocking_wall {
+                WallType::Clear => MoveValidity::Valid,
+                WallType::Impassable => {
+                    MoveValidity::Invalid("Can't pass through impassable wall".to_string())
+                }
+                _wildcard => Self::door_matches_heister(blocking_wall, heister_color),
```

Comment:
This isn't really valid because the tile may not have been placed at the door. I think we want to consider doors as impassable, and then change them to clear when we place the tile at the door.

Reply:
Keeping the helper; but going to have it return Invalid("moving through doors to un-placed tiles is not implmeented")